service/src/main/java/com/carhouse/service/impl/CarSaleServiceImpl.java

```diff
@@ -66,29 +73,43 @@ public CarSale getCarSale(final int carSaleId) {
     @Override
     public Integer addCarSale(final CarSale carSale) {
         LOGGER.debug("method addCarSale with parameter: [{}]", carSale);
-        return carSaleDao.addCarSale(carSale);
+        try {
+            return carSaleDao.addCarSale(carSale);
+        } catch (DataIntegrityViolationException ex) {
+            throw new WrongReferenceException("there is wrong references in your car sale");
+        }
     }
 
     /**
      * Update car sale.
      * Gets car sale id from carSale object
      *
      * @param carSale the car sale
+     * @return check or car sale is updated
      */
     @Override
-    public void updateCarSale(final CarSale carSale) {
+    public boolean updateCarSale(final CarSale carSale) {
         LOGGER.debug("method updateCarSale with parameter: [{}]", carSale);
-        carSaleDao.updateCarSale(carSale);
+        getCarSale(carSale.getCarSaleId());
+        try {
+            return carSaleDao.updateCarSale(carSale);
+        } catch (DataIntegrityViolationException ex) {
+            throw new WrongReferenceException("there is wrong references in your car sale");
+        }
     }
 
     /**
      * Delete car sale by id.
      *
      * @param carSaleId the car sale id
+     * @return check or car sale is deleted
      */
     @Override
-    public void deleteCarSale(final int carSaleId) {
+    public boolean deleteCarSale(final int carSaleId) {
         LOGGER.debug("method deleteCarSale with parameter: [{}]", carSaleId);
-        carSaleDao.deleteCarSale(carSaleId);
+        if (!carSaleDao.deleteCarSale(carSaleId)) {
+            throw new NotFoundException("car sale you try to delete does not exist");
```

Comment:
more information here should be provided. What id?

Reply:
I've added it